src/Service/Storage/Ef/EfGlobalGlobalStorage.cs

```diff
@@ -50,4 +50,44 @@ public async Task<int> UpdatePeriodicCredentialReportsAsync()
 
         return rows;
     }
+
+    /// <summary>
+    /// Records the amount of daily or weekly active users per tenant.
+    /// </summary>
+    /// <returns></returns>
+    public async Task<int> UpdatePeriodicActiveUserReportsAsync()
+    {
+        var result = _db.AccountInfo
+            .GroupJoin(
+                _db.Credentials,
+                accountInfo => accountInfo.Tenant,
+                credential => credential.Tenant,
+                (accountInformation, credentials) => new PeriodicActiveUserReport
+                {
+                    Tenant = accountInformation.Tenant,
+
+                    DailyActiveUsersCount = credentials
+                        .Where(x => x.LastUsedAt >= _timeProvider.GetUtcNow().UtcDateTime.AddDays(-1))
+                        .Select(x => x.UserId)
+                        .Distinct()
+                        .Count(),
+
+                    WeeklyActiveUsersCount = credentials
+                        .Where(x => x.LastUsedAt >= _timeProvider.GetUtcNow().UtcDateTime.AddDays(-7))
+                        .Select(x => x.UserId)
+                        .Distinct()
+                        .Count(),
+
+                    TotalUsersCount = credentials
+                        .Select(x => x.UserId)
+                        .Distinct()
+                        .Count(),
+
+                    CreatedAt = DateOnly.FromDateTime(_timeProvider.GetUtcNow().UtcDateTime)
+                });
+
+        await _db.PeriodicActiveUserReports.AddRangeAsync(result);
```

Comment:
Apparently the `Async` version of this method is only to allow special value generators to access the db. Can probably switch it to the sync version.